### PR TITLE
add CF CLI plugin for VMware Tanzu GemFire

### PR DESCRIPTION
**Pull request by jinmeiliao**
Thank you for contributing to the CF CLI Plugin Repository!

This repo contains both the plugin repo server and the metadata for CLI
community plugins.
Some of the requirements for PRs will apply to only one of these parts.

We're not allowed to accept any PRs without a signed CLA, no matter how small.
If your contribution falls under a company CLA but your membership is not public, expect delays while we confirm.

# Submitting Plugins

If you haven't yet, please review our contributing guidelines:  
https://github.com/cloudfoundry/cli-plugin-repo#submitting-plugins

In particular ensure the following requirements are being met:
* [x] The plugin's `name` field in `repo-index.yml` matches the `Name` field in the plugin's `plugin.PluginMetadata` section
* [x] The plugin's `url` field in `repo-index.yml` contains the same version from the `version` field

# Submitting PRs to CLIPR (the CLI Plugin Repo server)

All new code requires tests to protect against regressions.

## Description of the Change

We must be able to understand the design of your change from this description.
Keep in mind that the maintainer reviewing this PR may not be familiar with or
have worked with the code here recently, so please walk us through the concepts.

## Why Is This PR Valuable?

What benefits will be realized by the code change? What users would want this change? What user need is this change addressing?

## Applicable Issues

List any applicable Github Issues here

## How Urgent Is The Change?

Is the change urgent? If so, explain why it is time-sensitive.

## Other Relevant Parties

Who else is affected by the change?

### CODE REVIEW

File: repo-index.yml [hunk 1]
```diff
@@ -1834,3 +1834,24 @@ plugins:
...
   updated: 2015-07-02T00:00:00Z
   version: 1.0.0
+- authors:
+    - contact: abaker@pivotal.io
+      homepage: https://github.com/gemfire/tanzu-gemfire-management-cf-plugin
+      name: Pivotal
+  binaries:
+    - checksum: 08de98c38fff2ad0c5fde51f37ad9c1b
+      platform: osx
+      url: https://github.com/gemfire/tanzu-gemfire-management-cf-plugin/releases/download/v1.0.1/gemfire-osx
```
Comment (jinmeiliao) [inline repo-index.yml]: does the name of the executable have to the the same as the name of the "plugin"?

**Comment by JenGoldstrich** [comment]
Hey @jinmeiliao thanks for submitting this PR!

For us to merge this PR we have to get the travis build to pass, there are two failures in it that we are seeing.

The first failure is because the checksum you have provided is the md5 checksum of this binary and not the sha1 checksum (we validated this by pulling down the binary and checking its md5 checksum, it matched what you had put in the repo-index.yml.

The other failure is due to where your plugin was added in repo-index.yml, it has to be added alphabetically with the other plugins.

You will also need to sign EasyCLA, see above.

Please feel free to reach out if you need any help fixing either of these issues :)!

Jenna and @heyjcollins

**Comment by jinmeiliao** [comment]
> Hey @jinmeiliao thanks for submitting this PR!
> 
> For us to merge this PR we have to get the travis build to pass, there are two failures in it that we are seeing.
> 
> The first failure is because the checksum you have provided is the md5 checksum of this binary and not the sha1 checksum (we validated this by pulling down the binary and checking its md5 checksum, it matched what you had put in the repo-index.yml.
> 
> The other failure is due to where your plugin was added in repo-index.yml, it has to be added alphabetically with the other plugins.
> 
> You will also need to sign EasyCLA, see above.
> 
> Please feel free to reach out if you need any help fixing either of these issues :)!
> 
> Jenna and @heyjcollins

Hi,Jenna, thanks for replying. I will get these fixed soon. The two things I do need your help on are:
1. I believe I have signed the EasyCLA as an individual. but somehow the check still fails.
2. regarding the name of the executables, since we have two "gemfire" executable on different platforms, we had to add a suffix to differentiate it, would this be a problem? Does the name of the plugin have to match the executable?

**Comment by acrmp** [comment]
Hi @jinmeiliao,

1. We're not sure why EasyCLA is complaining. I would expect you to be covered by the Pivotal corporate contributor membership. Perhaps you need to publish an additional org membership on your GitHub profile?
2. A suffix on the name of the binary should be fine. The filename does not need to match the plugin name in the `repo-index.yml`. The `name` of the plugin in the `repo-index.yml` does need to match the name in the go code `PluginMetadata`. I believe it does for your plugin.

Thanks,

Andrew and @nickjameswebb

**Comment by jinmeiliao** [comment]
> Hi @jinmeiliao,
> 
> 1. We're not sure why EasyCLA is complaining. I would expect you to be covered by the Pivotal corporate contributor membership. Perhaps you need to publish an additional org membership on your GitHub profile?
> 2. A suffix on the name of the binary should be fine. The filename does not need to match the plugin name in the `repo-index.yml`. The `name` of the plugin in the `repo-index.yml` does need to match the name in the go code `PluginMetadata`. I believe it does for your plugin.
> 
> Thanks,
> 
> Andrew and @nickjameswebb

I just pushed another update, hopefully that would fix all checks. now the only thing is the CLA authorization. I am part of "pivotal-cf", "apache" and "gemfire", not sure how to add another organization or which organization that I should add.

**Comment by acrmp** [comment]
@jinmeiliao I think you will need to publish your organization membership for EasyCLA to be able to see it, see the GitHub docs here for how to do this:
https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership

Also the PR is now failing the tests in Travis because the YAML is malformed, looks like an indentation issue.

**Comment by jinmeiliao** [comment]
> @jinmeiliao I think you will need to publish your organization membership for EasyCLA to be able to see it, see the GitHub docs here for how to do this:
> https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership
> 
> Also the PR is now failing the tests in Travis because the YAML is malformed, looks like an indentation issue.

I did publicize my memberships, but it still complains.
Also, I fixed the indentation, now it failed at checksum. I generated these checksum using `shasum` command, not sure why it's failing.

**Comment by acrmp** [comment]
@jinmeiliao I think that last test failure may be a little misleading. I think you've provided the shasum correctly, but the test is also executing your plugin to validate that the name in the repo index yaml matches the name declared in your `PluginMetadata` in the go source code.

I think it's being tripped up by this conditional in your plugin here:
https://github.com/gemfire/tanzu-gemfire-management-cf-plugin/blob/master/cmd/main/main.go#L40

**Comment by jinmeiliao** [comment]
> @jinmeiliao I think that last test failure may be a little misleading. I think you've provided the shasum correctly, but the test is also executing your plugin to validate that the name in the repo index yaml matches the name declared in your `PluginMetadata` in the go source code.
> 
> I think it's being tripped up by this conditional in your plugin here:
> https://github.com/gemfire/tanzu-gemfire-management-cf-plugin/blob/master/cmd/main/main.go#L40

Hi, thanks for the tip, that condition is trying to see how the program is invoked, and it is supposed to return true if its invoked as a cf plugin,  We do have this tested in pipeline that installs it as a cf plugin and executes it and everything was working. 
https://concourse.gemfire-ci.info/teams/main/pipelines/tanzu-gemfire-management-cf-plugin/jobs/test-cloudcache-management-cf-pcc-1.11/builds/5

the `PluginMetadata` is defined here, seems correct.
https://github.com/gemfire/tanzu-gemfire-management-cf-plugin/blob/2082339137bf1a726f95b3729333db754db8b001/impl/gemfire/plugin.go#L81

Anything that pops in your mind that might be wrong?

**Comment by jinmeiliao** [comment]
> @jinmeiliao I think that last test failure may be a little misleading. I think you've provided the shasum correctly, but the test is also executing your plugin to validate that the name in the repo index yaml matches the name declared in your `PluginMetadata` in the go source code.
> 
> I think it's being tripped up by this conditional in your plugin here:
> https://github.com/gemfire/tanzu-gemfire-management-cf-plugin/blob/master/cmd/main/main.go#L40

@acrmp the travis test keeps failing, looking at where it failed, does that mean the executable's name has to match the plugin name?
https://github.com/cloudfoundry/cli-plugin-repo/blob/e4a640125d654b5d7eef2acdab7c7f0fb0fd3028/repo_index_test.go#L62

**Pull request merged**